### DISCUSSION TITLE
Sharing a static state between test cases

### DISCUSSION THREAD

**Discussion by 1g0rkR** [nextest-rs/nextest#1729] [Q&A]
Hi there, I would appreciate if somebody can explain me how to use a shared state in nextest ?
Here is some explanation.
My static state in /tests/common/config.rs
```
pub static CONFIGURATION: Lazy<Mutex<Configuration>> =
    Lazy::new(|| Mutex::new(Configuration::default()));
```
here is my integration_test.rs in /tests
```
pub mod test_suite_1_login {

    use crate::common::{config::CONFIGURATION, test_suites};

    #[tokio::test]
    async fn test_case_login_success() {
        let mut conf = CONFIGURATION.lock().unwrap();
        test_suites::login::test_case_login_success(&mut conf).await;
    }

    #[tokio::test]
    async fn test_case_login_wrong_credentials() {
        let mut conf = CONFIGURATION.lock().unwrap();
        println!("\n\nnnnnnn {:?} \n\n", *conf);
        test_suites::login::test_case_login_wrong_credentials(&mut conf).await;
    }
}
```
Nextest.toml config:
```
[profile.integration_test]
retries = 0
failure-output = "final"
fail-fast = false
test-threads = 1
threads-required = 1
final-status-level = "flaky"
```

cargo test --test integration_test -- --test_threads=1 --nocapture works well. I can initialize my static state in first test case, modify `conf` and use already initialized conf in next test case. 
Nextest somehow shows always "empty" initialized conf.

**Comment by sunshowers** [comment]
Hi - nextest is process-per-test and this is a fundamental limitation with it: https://nexte.st/docs/design/how-it-works/

If you can serialize the state to disk one option is to use a setup script: https://nexte.st/docs/configuration/setup-scripts/

(As a general principle I would avoid modifying cached state across tests by the way, that can lead to strange problems down the line. Ideally each test is as isolated as possible. In your case it may make sense to combine the two tests into one.)
